Modules/Devices/src/Devices.Application/Identities/Commands/ApproveDeletionProcess/Handler.cs

```diff
@@ -46,7 +47,8 @@ public async Task<ApproveDeletionProcessResponse> Handle(ApproveDeletionProcessC
 
         _eventBus.Publish(new TierOfIdentityChangedDomainEvent(identity, oldTierId, newTierId));
 
-        await _notificationSender.SendNotification(identity.Address, new DeletionProcessApprovedNotification(deletionProcess.GracePeriodEndsAt?.DaysUntilDate() ?? 0), cancellationToken);
+        var daysUntilDeletion = deletionProcess.GracePeriodEndsAt?.DaysUntilDate() ?? throw new DomainException(DomainErrors.GracePeriodHasNotYetExpired());
```

Comment:
Actually, if `GracePeriodEndsAt` is null at this point, then it is an internal server error. So instead of throwing a DomainException (which would result in a http status 400), we should just throw an Exception with a descriptive error message.